packages/synapse-react-client/src/components/AddToDownloadCart/AddToDownloadCart.tsx

```diff
@@ -0,0 +1,96 @@
+import { useState, useEffect } from 'react'
+import { Button } from '@mui/material'
+import { useSynapseContext } from '../../utils'
+import { FolderDownloadConfirmation } from '../download_list/FolderDownloadConfirmation'
+import { TableQueryDownloadConfirmation } from '../download_list/TableQueryDownloadConfirmation'
+import { DEFAULT_PAGE_SIZE } from '../../utils/SynapseConstants'
+import { SynapseConstants } from '../../utils'
+import { QueryBundleRequest } from '@sage-bionetworks/synapse-types'
+import { QueryWrapper } from '../QueryWrapper'
+import { QueryWrapperErrorBoundary } from '../QueryWrapperErrorBoundary'
+import SynapseClient from '../../synapse-client'
+import { GetAppTwoTone } from '@mui/icons-material'
+
+export type AddToDownloadCartProps = {
+  folderId: string
+}
+
+export const AddToDownloadCart: React.FC<AddToDownloadCartProps> = ({
+  folderId,
+}: AddToDownloadCartProps) => {
+  const { accessToken } = useSynapseContext()
+  const [showConfirmation, setShowConfirmation] = useState(false)
+  const [entityType, setEntityType] = useState<string | undefined>(undefined)
+
+  useEffect(() => {
+    const getEntity = async () => {
+      try {
+        // will return 40X if not logged in
+        const { concreteType: entityType } = await SynapseClient.getEntity(
+          accessToken,
+          folderId,
+        )
+        setEntityType(entityType)
+      } catch (e) {
+        console.error('Error on getting entity information = ', e)
+        // should this error out or just default to folder?
+        //setEntityType('org.sagebionetworks.repo.model.Folder');
+      }
+    }
+    getEntity()
+  }, [folderId, accessToken])
+
+  const handleClose = () => {
+    setShowConfirmation(false)
+  }
+
+  const onAddClick = () => {
+    setShowConfirmation(true)
+  }
+  const initQueryRequest: QueryBundleRequest = {
+    concreteType: 'org.sagebionetworks.repo.model.table.QueryBundleRequest',
+    entityId: folderId,
+    query: {
+      sql: `select * from ${folderId}`,
+      limit: DEFAULT_PAGE_SIZE,
+      sort: undefined,
+      additionalFilters: undefined,
+    },
+    partMask: SynapseConstants.BUNDLE_MASK_QUERY_RESULTS,
+  }
+  const queryWrapperKey = JSON.stringify(initQueryRequest)
+
+  function folderTsx() {
+    return (
+      <FolderDownloadConfirmation folderId={folderId} fnClose={handleClose} />
+    )
+  }
+
+  function tableTsx() {
+    return (
+      <QueryWrapper initQueryRequest={initQueryRequest} key={queryWrapperKey}>
+        <QueryWrapperErrorBoundary>
+          <TableQueryDownloadConfirmation />
+        </QueryWrapperErrorBoundary>
+      </QueryWrapper>
+    )
+  }
```

Comment:
3 issues with this pattern:
1. These are function components, so they should be TitleCase
2. Function components [should always be rendered using JSX](https://react.dev/reference/rules/react-calls-components-and-hooks#never-call-component-functions-directly) `<MyComponent />`, not function calls.
3. Once you make these changes, you'll be breaking the [no-unstable-nested-components](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-unstable-nested-components.md) rule, so you'll need to refactor them out of this component (in the same file is fine)